model/annotations/task_annotations.go

```diff
@@ -137,3 +137,40 @@ func RemoveSuspectedIssueFromAnnotation(taskId string, execution int, issue Issu
 		bson.M{"$pull": bson.M{SuspectedIssuesKey: issue}},
 	)
 }
+
+func UpdateAnnotation(taskId string, execution int, a *TaskAnnotation, source *Source) error {
+
+	update := bson.M{
+		TaskIdKey:        taskId,
+		TaskExecutionKey: execution,
+	}
+
+	if a.Metadata != nil {
+		update[MetadataKey] = a.Metadata
+	}
+	if a.Note != nil {
+		a.Note.Source = source
+		update[NoteKey] = a.Note
+	}
+	if a.Issues != nil {
```

Comment:
Just to be safe I would instead check that len(a.Issues) > 0 (same with suspected)